src/pocketmine/Player.php

```diff
@@ -3171,11 +3171,13 @@ public function transfer(string $address, int $port = 19132, string $message = "
 	 *
 	 * @param string $reason
 	 * @param bool   $isAdmin
+	 * @param TextContainer|string $quitMessage
 	 *
 	 * @return bool
 	 */
-	public function kick(string $reason = "", bool $isAdmin = true) : bool{
-		$this->server->getPluginManager()->callEvent($ev = new PlayerKickEvent($this, $reason, $this->getLeaveMessage()));
+	public function kick(string $reason = "", bool $isAdmin = true, $quitMessage = null) : bool{
+		$quitMessage = $quitMessage ?? $this->getLeaveMessage();
+		$this->server->getPluginManager()->callEvent($ev = new PlayerKickEvent($this, $reason, $quitMessage));
```

Comment:
You could just pass `$quitMessage ?? $this->getLeaveMessage()` directly to `PlayerKickEvent` without overwriting the variable, since it isn't used anywhere else.